### DISCUSSION TITLE
Improved Prompt Handling in Open Web UI

### DISCUSSION THREAD

**Discussion by frettig-radiox** [open-webui/open-webui#5024] [Ideas]
#### Problem:

When a message is sent to the LLM that starts with "/", a selection of prompts on various topics is suggested. The selection shows the title of the prompt, but not the text. To see the text of the prompt, the selection must first be confirmed. The prompt text may contain placeholders for variables in square brackets and `{{CLIPBOARD}}`. The clipboard placeholder is replaced directly when selecting a prompt. The problem is that the user only sees this afterward, for example, when a very long text (which may have still been in the clipboard) is inserted.

#### Suggested Improvements:

1. The `{{CLIPBOARD}}` placeholder should only be replaced when the prompt is submitted. This allows the user to update the clipboard after reading the complete prompt.

2. An icon could indicate that the prompt intends to access the clipboard (see graphic).

3. In the text input, the prompt text could be displayed as a preview when a prompt is preselected (see graphic).

4. Similar to the clipboard icon, variables in the prompt could also be announced via an icon (see graphic). These could be displayed in advance using a tooltip.

<img width="649" alt="prompt_dialog" src="https://github.com/user-attachments/assets/14f21b40-58d9-414d-a23a-20092ba83b52">

**Comment by havejoe** [comment]
I fully support this feature request. The ability to view the full prompt text and understand what content will be inserted before submitting it is crucial for a smoother user experience. Delaying the clipboard content insertion and providing clear indicators for clipboard access and variables, would make the interface much more intuitive
